examples/basic/ipc_shm_ipcapi_consumer.c

```diff
@@ -0,0 +1,229 @@
+
+    // set the IP address and the port
+    consumer_addr.sin_family = AF_INET;
+    consumer_addr.sin_port = htons(port);
+    consumer_addr.sin_addr.s_addr = inet_addr(INET_ADDR);
+
+    // bind to the IP address and the port
+    if (bind(consumer_socket, (struct sockaddr *)&consumer_addr,
+             sizeof(consumer_addr)) < 0) {
+        fprintf(stderr, "[consumer] ERROR: cannot bind to the port\n");
+        goto err_close_consumer_socket;
+    }
+
+    fprintf(stderr, "[consumer] Binding done\n");
+
+    // listen for the producer
+    if (listen(consumer_socket, 1) < 0) {
+        fprintf(stderr, "[consumer] ERROR: listen() failed\n");
+        goto err_close_consumer_socket;
+    }
+
+    fprintf(stderr, "[consumer] Listening for incoming connections ...\n");
+
+    // accept an incoming connection
+    producer_addr_len = sizeof(producer_addr);
+    producer_socket = accept(consumer_socket, (struct sockaddr *)&producer_addr,
+                             (socklen_t *)&producer_addr_len);
+    if (producer_socket < 0) {
+        fprintf(stderr, "[consumer] ERROR: accept() failed\n");
+        goto err_close_consumer_socket;
+    }
+
+    fprintf(stderr, "[consumer] Producer connected at IP %s and port %i\n",
+            inet_ntoa(producer_addr.sin_addr), ntohs(producer_addr.sin_port));
+
+    ret = producer_socket; // success
+
+err_close_consumer_socket:
+    close(consumer_socket);
+
+    return ret;
+}
+
+int main(int argc, char *argv[]) {
+    char consumer_message[CONSUMER_MSG_SIZE];
+    char recv_buffer[RECV_BUFF_SIZE];
+    int producer_socket;
+    int ret = -1;
+
+    if (argc < 2) {
+        fprintf(stderr, "usage: %s port\n", argv[0]);
+        return -1;
+    }
+
+    int port = atoi(argv[1]);
+
+    umf_memory_provider_handle_t OS_memory_provider = NULL;
+    umf_os_memory_provider_params_t os_params;
+    enum umf_result_t umf_result;
+
+    os_params = umfOsMemoryProviderParamsDefault();
+    os_params.visibility = UMF_MEM_MAP_SHARED;
+
+    // create OS memory provider
+    umf_result = umfMemoryProviderCreate(umfOsMemoryProviderOps(), &os_params,
+                                         &OS_memory_provider);
+    if (umf_result != UMF_RESULT_SUCCESS) {
+        fprintf(stderr,
+                "[consumer] ERROR: creating OS memory provider failed\n");
+        return -1;
+    }
+
+    umf_memory_pool_handle_t scalable_pool;
+    umf_result = umfPoolCreate(umfScalablePoolOps(), OS_memory_provider, NULL,
+                               0, &scalable_pool);
+    if (umf_result != UMF_RESULT_SUCCESS) {
+        fprintf(stderr,
+                "[producer] ERROR: creating jemalloc UMF pool failed\n");
+        goto err_destroy_OS_memory_provider;
+    }
+
+    producer_socket = consumer_connect(port);
+    if (producer_socket < 0) {
+        goto err_destroy_scalable_pool;
+    }
+
+    // receive the IPC handle from the producer's
+    ssize_t len = recv(producer_socket, recv_buffer, RECV_BUFF_SIZE, 0);
```

Comment:
what if the size of the IPC handle is greater than `RECV_BUFF_SIZE`?

I think we need to send the size of the IPC handle first, and then receive the exact amount of bytes corresponding to IPC handle. It will make our example clearer and better demonstrate how to transfer IPC handle between processes.

Reply:
Done